common/lib/xmodule/xmodule/x_module.py

```diff
@@ -1743,6 +1744,35 @@ def resource_url(self, resource):
     def publish(self, block, event_type, event):
         pass
 
+    def service(self, block, service_name):
+        """
+        Runtime-specific override for the XBlock service manager.  If a service is not currently
+        instantiated and is declared as a critical requirement, an attempt is made to load the
+        module.
+
+        Arguments:
+            block (an XBlock): this block's class will be examined for service
+                decorators.
+            service_name (string): the name of the service requested.
+
+        Returns:
+            An object implementing the requested service, or None.
+        """
+        try:
+            service = super(ModuleSystem, self).service(block=block, service_name=service_name)
+        except NoSuchServiceError:
+            from xmodule.modulestore.django import ModuleI18nService
```

Comment:
Recommendation: don't use local imports. Instead, check `_service` for a the requested service name, and then use `callable(service)` to find out if it's callable. If it is, return `service(xblock)`, otherwise, return `super(...).service(block=block, service_name=service_name)`. Then, pass `ModuleI18nService` in with the rest of the services in LMS and Studio, but as a class, rather than an instantiated object.